### ISSUE TITLE
Update plugin description

### ISSUE THREAD

**Issue by googleson78** [polysemy-research/polysemy#213]
Right now the plugin is also responsible for some of "making polysemy go fast", if I understand correctly. (getting this info from the "Stellar Engineering" section and also `Phases.hs`)

Maybe we could update its `README.md` to make that clear as it's what shows up as documentation on `hackage`, and right now one would think it's only a tool for convenience. I would do so, but I don't really know what to write there.

**Comment by isovector** [comment]
Yeah, good idea. The plugin enables some internal core optimizations that turn `-O2` into the mythical `-O3`.
